### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by pavel-gutsal**
https://pavel-gutsal.github.io/layout_calendar/

div elements manualy created are used as days. 
@each cycle (loop) wasn`t found to be useful in my case
saturday and sunday background was changed by pseudo class :nth-child(n)
due to div elements manual creation , modifier `month-length` wasn`t used
if I made something wrong - please provide correct example.

### CODE REVIEW

**Comment by vkryvytskyi** [comment]
you should add for the days wrapper three classes
```html
<div class='calendar calendar--28 calendar--wed'>
```
this would mean that month start at Wednesday and has 28 days. And so needed styles would be applied depending on values in this modifiers

**Review by vkryvytskyi — CHANGES_REQUESTED** [review]
add both modifier-classes to the days wrapper element

**Comment by pavel-gutsal** [comment]
main.scss was complitly rewritten . Was found an issue when 
@for $index from 28 through 31 {
  .calendar--#{$index} .calendar__day:nth-child(n+30 ) {
    display: flex;
  }

}
is not working as expected .(specificly nth-child(n+30 ))


hacky solution was  to  give all elements display:none , then to write  nth-child( -n+#{$index} )  - where all child elements are diven display:flex;

File: src/styles/main.scss [hunk 1]
```diff
@@ -1,3 +1,80 @@
...
+        content: "#{$ind}";
+        font-size: 30px;
+        font-family: Arial, Helvetica, sans-serif;
+      }
+
+    }
+
+  }
+
+}
```
Comment (marina-tilniak) [inline src/styles/main.scss]:
```suggestion
    @for $ind from 1 through 31 {
      &:nth-child(#{$ind})::before {
        content: "#{$ind}";
        font-size: 30px;
        font-family: Arial, Helvetica, sans-serif;
      }
    }
  }
}
```

File: src/styles/main.scss [hunk 2]
```diff
@@ -1,3 +1,80 @@
...
+$padding:10px;
+$calendarWidth:$padding*2 + $gap * 6 + $boxSize * 7;
+$margins: ( 
+  "mon":0 , 
+  "tue": 101px , 
+  "wed":202px , 
+  "thu":303px , 
+  "fri":404px , 
+  "sat":505px , 
+  "sun":606px
```
Comment (marina-tilniak) [inline src/styles/main.scss]: use your variables

File: src/styles/main.scss [hunk 3]
```diff
@@ -1,3 +1,80 @@
...
+  .calendar--#{$key} .calendar__day:first-child {
+    margin-left: $value;
+  }
+
+}
+
+@for $index from 28 through 31 {
+  .calendar--#{$index} .calendar__day:nth-child( -n+#{$index} ) {
+    display: flex;
+  }
```
Comment (marina-tilniak) [inline src/styles/main.scss]:
```suggestion
@for $index from 28 through 31 {
  .calendar--#{$index} .calendar__day:nth-child( n+#{$index + 1} ) {
    display: none;
  }
```

File: src/styles/main.scss [hunk 4]
```diff
@@ -1,3 +1,80 @@
...
+  gap: $gap;
+  padding: $padding;
+  width: $calendarWidth;
+
+  &__day {
+    width: $boxSize;
+    height: $boxSize;
+    background: #eee;
+    border: 1px solid black;
+    display: none;
```
Comment (marina-tilniak) [inline src/styles/main.scss]:
```suggestion
    display: flex;
```

**Review by marina-tilniak — CHANGES_REQUESTED** [review]

**Comment by pavel-gutsal** [comment]
all issues described above were fixed

**Review by ArturErmolenko — APPROVED** [review]

**Pull request closed without merging**